trading_calendars/exchange_calendar_xssc.py

```diff
@@ -0,0 +1,497 @@
+    "2016-06-10",
+    "2016-09-15",
+    "2016-09-16",
+    "2016-10-03",
+    "2016-10-04",
+    "2016-10-05",
+    "2016-10-06",
+    "2016-10-07",
+    "2017-01-02",
+    "2017-01-27",
+    "2017-01-30",
+    "2017-01-31",
+    "2017-02-01",
+    "2017-02-02",
+    "2017-04-03",
+    "2017-04-04",
+    "2017-05-01",
+    "2017-05-29",
+    "2017-05-30",
+    "2017-10-02",
+    "2017-10-03",
+    "2017-10-04",
+    "2017-10-05",
+    "2017-10-06",
+    "2018-01-01",
+    "2018-02-15",
+    "2018-02-16",
+    "2018-02-19",
+    "2018-02-20",
+    "2018-02-21",
+    "2018-04-05",
+    "2018-04-06",
+    "2018-04-30",
+    "2018-05-01",
+    "2018-06-18",
+    "2018-09-24",
+    "2018-10-01",
+    "2018-10-02",
+    "2018-10-03",
+    "2018-10-04",
+    "2018-10-05",
+    "2019-01-01",
+    "2019-02-04",
+    "2019-02-05",
+    "2019-02-06",
+    "2019-02-07",
+    "2019-02-08",
+    "2019-04-05",
+    "2019-05-01",
+    "2019-06-07",
+    "2019-09-13",
+    "2019-10-01",
+    "2019-10-02",
+    "2019-10-03",
+    "2019-10-04",
+    "2019-10-07",
+    "2020-01-01",
+    "2020-01-27",
+    "2020-01-28",
+    "2020-01-29",
+    "2020-01-30",
+    "2020-05-01",
+    "2020-06-25",
+    "2020-10-01",
+    "2020-10-02",
+    "2020-10-05",
+    "2021-01-01",
+    "2021-02-12",
+    "2021-02-15",
+    "2021-02-16",
+    "2021-02-17",
+    "2021-04-05",
+    "2021-06-14",
+    "2021-09-21",
+    "2021-10-01",
+    "2021-10-04",
+    "2021-10-05",
+    "2022-01-03",
+    "2022-02-01",
+    "2022-02-02",
+    "2022-02-03",
+    "2022-02-04",
+    "2022-04-05",
+    "2022-06-03",
+    "2022-10-03",
+    "2022-10-04",
+    "2022-10-05",
+    "2023-01-02",
+    "2023-01-23",
+    "2023-01-24",
+    "2023-01-25",
+    "2023-01-26",
+    "2023-01-27",
+    "2023-04-05",
+    "2023-05-01",
+    "2023-06-22",
+    "2023-09-29",
+    "2023-10-02",
+    "2023-10-03",
+    "2023-10-04",
+    "2023-10-05",
+    "2024-01-01",
+    "2024-02-12",
+    "2024-02-13",
+    "2024-02-14",
+    "2024-02-15",
+    "2024-04-04",
+    "2024-05-01",
+    "2024-06-10",
+    "2024-09-17",
+    "2024-10-01",
+    "2024-10-02",
+    "2024-10-03",
+    "2024-10-04",
+    "2025-01-01",
+    "2025-01-29",
+    "2025-01-30",
+    "2025-01-31",
+    "2025-02-03",
+    "2025-04-04",
+    "2025-05-01",
+    "2025-10-01",
+    "2025-10-02",
+    "2025-10-03",
+    "2025-10-04",
+])
+
+start_default = pd.Timestamp("1999-01-01", tz='UTC')
+end_default = pd.Timestamp("2025-10-04", tz='UTC')
+
+
+class XSSCExchangeCalendar(PrecomputedTradingCalendar):
+    """
+    Exchange calendar for the Shanghai Stock Exchange (XSSC, SSE).
+
+    Open time: 9:30 Asia/Shanghai
+    Close time: 15:00 Asia/Shanghai
+
+    NOTE: For now, we are skipping the intra-day break from 11:30 to 13:00.
+
+    Due to the complexity around the BSE holidays, we are hardcoding a list
+    of holidays covering 1999-2025, inclusive. There are no known early closes
+    or late opens.
+    """
+
+    name = "XSSC"
```

Comment:
I don't know for sure, but should this be `XSHG`? (just going by https://www.iso20022.org/sites/default/files/ISO10383_MIC/ISO10383_MIC.pdf)